kernel/term_typing.ml

```diff
@@ -507,7 +514,9 @@ let translate_local_assum env t =
     t
 
 let translate_recipe env kn r =
-  build_constant_declaration kn env (Cooking.cook_constant env r)
+  let (_, dir, _) = Constant.repr3 kn in
+  let hcons = DirPath.is_empty dir in
+  build_constant_declaration kn env (Cooking.cook_constant ~hcons env r)
```

Comment:
Please document that we're hash consing only outside of sections, and why. Imagine an engineer reads this code.